examples/07_advanced/plot_mask_large_fmri.py

```diff
@@ -0,0 +1,279 @@
+
+mask_paths = []
+for i, mask in enumerate(iter_img(masks)):
+    resampled_mask = resample_to_img(
+        mask,
+        fmri_path,
+        interpolation="nearest",
+        copy_header=True,
+        force_resample=True,
+    )
+    path = output_dir / f"mask_{i}.nii.gz"
+    data = resampled_mask.get_fdata()
+    data[data != 0] = 1
+    resampled_mask = resampled_mask.__class__(
+        data, resampled_mask.affine, resampled_mask.header
+    )
+    resampled_mask.to_filename(path)
+    mask_paths.append(path)
+
+# %%
+# Mask the fMRI image using NiftiMasker
+# -------------------------------------
+# Let's first see how we would typically mask the fMRI image using
+# :class:`~nilearn.maskers.NiftiMasker`. This is the most common way to extract
+# data from an fMRI image as it makes it easy to standardize, smooth, detrend,
+# etc. the data.
+#
+# We will define a function that will do that so that it's easier to use it
+# with the ``memory_profiler`` package to measure the memory usage.
+
+from memory_profiler import memory_usage
+
+from nilearn.maskers import NiftiMasker
+
+
+def nifti_masker_single(fmri_path, mask_path):
+    return NiftiMasker(mask_img=mask_path).fit_transform(fmri_path)
+
+
+nifti_masker_single_usage = memory_usage(
+    (nifti_masker_single, (fmri_img, mask_paths[0])), max_usage=True
+)
+print(
+    "Peak memory usage for masking with a single mask using NiftiMasker:\n\n"
+    f"{nifti_masker_single_usage} MiB"
+)
+
+# %%
+# Masking using NiftiMasker in parallel
+# -------------------------------------
+# Now let's see how we would mask the fMRI image using multiple masks in
+# parallel using the :mod:`joblib` package.
+
+from joblib import Parallel, delayed
+
+
+def nifti_masker_parallel(fmri_path, mask_paths):
+    return Parallel(n_jobs=N_REGIONS)(
+        delayed(nifti_masker_single)(fmri_path, mask) for mask in mask_paths
+    )
+
+
+nifti_masker_parallel_usage = memory_usage(
+    (nifti_masker_parallel, (fmri_img, mask_paths)),
+    max_usage=True,
+    include_children=True,
+    multiprocess=True,
+)
+print(
+    f"Peak memory usage for masking with {N_REGIONS} jobs in parallel\n"
+    "using NiftiMasker:\n\n"
+    f"{nifti_masker_parallel_usage} MiB"
+)
+
+# %%
+# Masking using numpy indexing
+# ----------------------------
+# Now let's see how we can mask the fMRI image using numpy indexing. This
+# could be more efficient than the NiftiMasker when we simply need to mask
+# the fMRI image with binary masks and don't need to standardize, smooth, etc.
+# the image.
+#
+# In addition, we will use :func:`nibabel.loadsave.load` function to load the
+# fMRI image as a proxy object. This will allow us to load the data directly
+# from the file as a numpy array without loading the entire image into memory.
+# You can find more information about this in the :mod:`nibabel` documentation,
+# here: https://nipy.org/nibabel/images_and_memory.html
+
+import numpy as np
+from nibabel import load
```

Comment:
small but this is an unusual import ; can you just load `nibabel as nib` and call `nib.load` ? That's how we do it in [other](https://nilearn.github.io/stable/auto_examples/03_connectivity/plot_atlas_comparison.html#extract-coordinates-on-yeo-atlas-parcellations) [examples](https://nilearn.github.io/stable/auto_examples/04_glm_first_level/plot_bids_features.html#first-level-model-estimation-one-subject)